src/bin/js/worker_interface.ml

```diff
@@ -0,0 +1,995 @@
+(******************************************************************************)
+(*                                                                            *)
+(*     Alt-Ergo: The SMT Solver For Software Verification                     *)
+(*     Copyright (C) 2018-2020 --- OCamlPro SAS                               *)
+(*                                                                            *)
+(*     This file is distributed under the terms of the license indicated      *)
+(*     in the file 'License.OCamlPro'. If 'License.OCamlPro' is not           *)
+(*     present please contact us to clarify licensing.                        *)
+(*                                                                            *)
+(******************************************************************************)
+open Js_of_ocaml
+open Data_encoding
+
+(** Types extract from AltErgoLib Utils.util and Utils.options *)
+
+type model = MNone | MDefault | MAll | MComplete
+
```

Comment:
Not sure we should redefine all the types in options.ml here. We can directly use those defined in Options. This will (hopefully) allow to modify this file when types in options.ml evolve (thanks to the compiler)

Reply:
I choose to create worker_interface as independent as possible from the alt-ergo-lib options file. OCaml users of the webworker can easily get this helper file and use the functions to convert options to json file or recover results from a json file sent by the worker.

Put the type `options` in options.ml was an idea raised at the time of #302, to improve the usability of the Alt-Ergo-lib. This solution should need some discussion for a future improvement.